client/extensions/woocommerce/app/promotions/promotion-validations.js

```diff
@@ -0,0 +1,297 @@
+/** @format */
+// TODO: Add unit tests for the following functions.
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { isEmpty, isUndefined } from 'lodash';
+
+/**
+ * Validates a coupon code.
+ * @param { String } fieldName The field name to be validated.
+ * @param { Object } promotion The promotion to be validated.
+ * @param { String } currency The currency to be used for validation.
+ * @param { bool } showEmpty True if empty fields should result in a validation error.
+ * @return { string } Returns a validation error, or undefined if none.
+ */
+export function validateCouponCode( fieldName, promotion, currency, showEmpty ) {
+	if ( showEmpty && isEmpty( promotion.couponCode ) ) {
```

Comment:
Should this be `promotion[ fieldName ]`? Same Q for the other functions… you could also merge the `validatePercentDiscount` & `validateFixedDiscount` functions, since they seem identical. I see in most cases the unique part is the error string, so I understand not abstracting it totally, but it's a little misleading to pass fieldName in while hardcoding couponCode etc.

Reply:
On second thought - after reading through more of the functions - i'm not saying to change this, but it does feel strange to have the `fieldName` arg in all of these functions when it's only used in `validateStartEndDate`. I know why they all need the same signature, though, so I think it's fine in the end.

Reply:
Yeah, I was a bit iffy on the fieldName as well. I was thinking of moving it farther back in the parameter list to make it more optional. What do you think of that?

Reply:
In looking at the parameter order, I don't know that it would make much of a difference. It doesn't make sense to put `fieldName` after `showEmpty` or `currency`. I think I'll leave it as-is for now, but I'm definitely up for modifying this later.